src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/GenerateDocumentHandler.java

```diff
@@ -40,14 +44,18 @@ public boolean canHandle(
         requireNonNull(callbackStage, "callbackStage must not be null");
         requireNonNull(callback, "callback must not be null");
 
+        List<Event> allowedEvents = Lists.newArrayList(Event.SUBMIT_APPEAL, Event.LIST_CASE);
+        if (isEmStitchingEnabled) {
+            allowedEvents.add(Event.GENERATE_HEARING_BUNDLE);
+        }
+
         return isDocmosisEnabled
+               && isEmStitchingEnabled
                && callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-               &&
-               Arrays.asList(
-                   Event.SUBMIT_APPEAL,
-                   Event.LIST_CASE,
-                   Event.GENERATE_HEARING_BUNDLE
-               ).contains(callback.getEvent());
+               && allowedEvents.contains(callback.getEvent())
+               || isDocmosisEnabled
+                  && callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                  && allowedEvents.contains(callback.getEvent());
```

Comment:
This is nasty condition. Although I appreciate the formatting efforts to add clarity, Jamie would be proud!

Reply:
Yup it's not ideal.  The feature flag should go away in a couple of months so ok for now.